### PR TITLE
Username validation api#827

### LINKED ISSUE #827: [Feature Request|Bug] Enforcing Username Uniqueness and Character Restrictions for Account Management
Usernames should be unique; currently, admin and Admin are two separate accounts.
Usernames should not contain spaces or percentage signs - a user with spaces cannot be deleted, and further special characters need to be checked.

### PR DESCRIPTION
**Pull request by mannam11**
Added username validation for Api's which are processing username.

Closes #827 

## Checklist:

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Contributor License Agreement

By submitting this pull request, I acknowledge and agree that my contributions will be included in Stirling-PDF and that they can be relicensed in the future under the MPL 2.0 (Mozilla Public License Version 2.0) license.

(This does not change the general open-source nature of Stirling-PDF, simply moving from one license to another license)

### CODE REVIEW

**Comment by Frooodle** [comment]
Can you please stop making new PRs when you make a change, you can do all existing changes as part of same PR, so it keeps all comments and reviews tracked

File: src/main/java/stirling/software/SPDF/controller/api/UserController.java [hunk 1]
```diff
@@ -44,6 +40,14 @@ public class UserController {
     @PreAuthorize("!hasAuthority('ROLE_DEMO_USER')")
     @PostMapping("/register")
     public String register(@ModelAttribute UsernameAndPass requestModel, Model model) {
+
+        if (!userService.isUsernameValid(requestModel.getUsername())) {
+            model.addAttribute(
+                    "invalidUsername",
+                    "Invalid username, Username must only contain alphabet characters and numbers.");
```
Comment (Frooodle) [inline src/main/java/stirling/software/SPDF/controller/api/UserController.java]: We can not use this route for error messages as it doesnt enable them to be translated instead something like the messageType in other entries should be used
Since i dont think register actually is used i guess this is fine though, i wonder if we can remove this whole method.. I will look into this

File: src/main/resources/templates/change-creds.html [hunk 2]
```diff
@@ -7,73 +7,78 @@
...
     <th:block th:insert="~{fragments/common :: game}"></th:block>
     <div id="page-container">
-      <div id="content-wrap">
-        <th:block th:insert="~{fragments/navbar.html :: navbar}"></th:block>
-        <br /><br />
-        <div class="container">
-          <div class="row justify-content-center">
-            <div class="col-md-9">
+        <div id="content-wrap">
+            <div th:insert="~{fragments/navbar.html :: navbar}"></div>
```
Comment (Frooodle) [inline src/main/resources/templates/change-creds.html]: why was this changed from th:block to div?

Comment (mannam11) [reply src/main/resources/templates/change-creds.html]: Sorry for this change, I am working on it to correct the change.

File: src/main/java/stirling/software/SPDF/controller/api/UserController.java (hunk 1 above)
Comment (mannam11) [reply src/main/java/stirling/software/SPDF/controller/api/UserController.java]: Are we going to remove this Register method all together?

Comment (Frooodle) [reply src/main/java/stirling/software/SPDF/controller/api/UserController.java]: Maybe but you don't need to address this in your change

Comment (Ludy87) [reply src/main/java/stirling/software/SPDF/controller/api/UserController.java]: I wouldn't remove it, maybe sooner or later someone will still need it.

File: src/main/resources/templates/account.html [hunk 3]
```diff
@@ -7,274 +7,303 @@
...
+
+							</script> 
+                        
+                        <div class="mb-3 mt-4">
+                            <a href="logout">
+                                <button type="button" class="btn btn-danger" th:text="#{account.signOut}">Sign Out</button>
+                            </a>
+                            <a th:if="${role == 'ROLE_ADMIN'}" href="addUsers" target="_blank">
+						        <button type="button" class="btn btn-info"  th:text="#{account.adminSettings}">Admin Settings</button>
+						    </a>
```
Comment (sbplat) [inline src/main/resources/templates/account.html:306]: Why did the entire chunk get changed?

File: src/main/resources/templates/addUsers.html [hunk 4]
```diff
@@ -41,33 +41,42 @@ <h2 class="text-center" th:text="#{adminUserSettings.header}">Admin User Control
...
-                </div>
-                <div class="mb-3">
-                  <input type="checkbox" class="form-check-input" id="forceChange" name="forceChange">
-                  <label class="form-check-label" for="forceChange" th:text="#{adminUserSettings.forceChange}">Force user to change username/password on login</label>
-                </div>
-
-                <!-- Add other fields as required -->
-                <button type="submit" class="btn btn-primary" th:text="#{adminUserSettings.submit}">Save User</button>
-              </form>
+						<h2 th:text="#{adminUserSettings.addUser}">Add New User</h2>
```
Comment (sbplat) [inline src/main/resources/templates/addUsers.html:44]: Different indentation?

File: src/main/resources/templates/change-creds.html [hunk 5]
```diff
@@ -7,34 +7,38 @@
   <body>
     <th:block th:insert="~{fragments/common :: game}"></th:block>
     <div id="page-container">
-      <div id="content-wrap">
-        <th:block th:insert="~{fragments/navbar.html :: navbar}"></th:block>
-        <br /><br />
-        <div class="container">
-          <div class="row justify-content-center">
-            <div class="col-md-9">
+        <div id="content-wrap">
```
Comment (sbplat) [inline src/main/resources/templates/change-creds.html:10]: Same here. Could we use 2 spaces instead of 4 to preserve the style?

File: src/main/resources/messages_en_US.properties [hunk 6]
```diff
@@ -53,9 +53,7 @@ notAuthenticatedMessage=User not authenticated.
 userNotFoundMessage=User not found.
 incorrectPasswordMessage=Current password is incorrect.
 usernameExistsMessage=New Username already exists.
-deleteCurrentUserMessage=Cannot delete currently logged in user.
-deleteUsernameExistsMessage=The username does not exist and cannot be deleted.
-
+invalidUsernameMessage=Invalid username, Username must only contain alphabet characters and numbers.
```
Comment (sbplat) [inline src/main/resources/messages_en_US.properties:56]: Please add this to en_GB as well.

**Review by sbplat — CHANGES_REQUESTED** [review]

**Comment by Ludy87** [comment]
Please adhere to the HTML5 standard: https://google.github.io/styleguide/htmlcssguide.html#Indentation
